app/Services/GraphQL.php

```diff
@@ -15,14 +16,50 @@ public function __construct()
     }
 
     /**
-     * Run a GraphQL query using the client.
+     * Run a GraphQL query using the client and return the data result.
      *
      * @param  $query     String
      * @param  $variables Array
-     * @return \Softonic\GraphQL\Response
+     * @return array|null
      */
     public function query($query, $variables)
     {
-        return $this->client->query($query, $variables);
+        // Use try/catch to avoid any GraphQL related errors breaking the application.
+        try {
+            $response = $this->client->query($query, $variables);
+        } catch (\Exception $exception) {
+            Log::error('GraphQL request failed.', [
+                'query' => $query,
+                'variables' => $variables,
+                'exception' => $exception->getMessage(),
+            ]);
+
+            return null;
+        }
+
+        return $response ? $response->getData() : null;
+    }
+
+    /**
+     * Query for a CampaignWebsite by campaignId field.
+     *
+     * @param  $campaignId String
+     * @return array|null
+     */
+    public function getCampaignWebsiteByCampaignId($campaignId)
+    {
+        $query = '
+        query GetCampaignWebsiteByCampaignId($campaignId: String!) {
+          campaignWebsiteByCampaignId(campaignId: $campaignId) {
+            title
+            slug
+          }
+        }';
+
+        $variables = [
+            'campaignId' => $campaignId,
+        ];
+
+        return $this->query($query, $variables, 'campaignWebsiteByCampaignId');
```

Comment:
I think this needs to be updated after removing the `$queryKey` parameter in 45c6850:

```php
return $this->query($query, $variables)['campaignWebsiteByCampaignId'];
```

Reply:
🤦‍♂

Reply:
Would it make sense for the individual query methods to be parsing out the top level query key themselves, or should that responsibility rest with the invoking file? I'm thinking the reason the former might make sense is in order to be consistent with queries which do use multiple queries in one, where we wouldn't be able to just return the top level query key?

Reply:
Hmmm, yeah that's a fair point... 🤔 

Now that you mention it, I wonder if it would make more sense to actually move this whole query _out_ of the `GraphQL` class and into the specific places we use it (so in the job itself, like how we colocate queries with their respective components in Phoenix). That way you could easily reference or change the query in the same place it's being used...

Reply:
Would it be an issue tho that we'd have to repeat that query across the [Post](https://github.com/DoSomething/rogue/blob/bc9561140a8ed98f39a7c257feaa725b8e5e7d73/app/Models/Post.php#L210-L243) and [Signup](https://github.com/DoSomething/rogue/blob/bc9561140a8ed98f39a7c257feaa725b8e5e7d73/app/Models/Signup.php#L139-L160) `toBlinkPayload` methods (I think those are the locations you mean), since we'll be needing these campaign fields in both places?

Ohh or maybe we can add the logic for the query as a method on the Campaign model itself so it's more specific? Or for that price maybe it's just worth having it defined in the service...

I dunno, maybe based on the Phoenix component usage example this indeed _would_ be worth repeating across the various places as we'd do with two different components using the same query (though in that case would we define it once as a fragment)? (lol sorry I'm making this so confusing 😖 🙃I guess my mental model for gracefully implementing graphQL in an app like this needs some serious therapy.)

Reply:
I don't think there's a right answer, to be honest!

Philosophically, I think it does make sense to keep the queries as close to the place they're used since it makes it easier to understand & edit in the future… but I can also see how that'd make our (already pretty hefty) models even heftier!

I'm good with trying whichever approach you think works best for now, and we can always adjust and try other things if we end up making more GraphQL queries in this app.

Reply:
Fair! How's this for a compromise for the time being:
I'll leave the helper method since we're -as of now- using it identically across those models, but I'll also leave a comment specifying our preferred bundling approach so that this doesn't set a philosophically deviant precedent?

Reply:
Smart! 👍

Reply:
Added with https://github.com/DoSomething/rogue/pull/889/commits/8f95dfc20e107c471db42fd11aa5d629e84ea404!